packages/medusa/src/services/gift-card.ts

```diff
@@ -180,6 +181,30 @@ class GiftCardService extends TransactionBaseService {
     })
   }
 
+   /**
+   * The tax_rate of the giftcard can depend on whether regions tax gift cards, an input
+   * provided by the user or the tax rate. Based on these conditions, tax_rate changes.
+   * @return the tax rate for the gift card
+   */
+  protected static fetchTaxRate(
```

Comment:
suggestion(non blocking): `resolveTaxRate`, There is no notion of fetching something but more of resolving the tax rate wdyt?